abstractions/typescript/src/authentication/authenticationProvider.ts

```diff
@@ -1,11 +1,26 @@
 import { RequestInformation } from "../requestInformation";
 
-/** Authenticates the application request. */
+/**
+ * @interface
+ * Interface to be implementated to provide authentication information for a request.
+ * @property {Function} authenticateRequest - The function to authenticate the request.
+ */
 export interface AuthenticationProvider {
     /**
      * Authenticates the application and returns a token base on the provided Uri.
      * @param request the request to authenticate.
      * @return a Promise to await for the authentication to be completed.
      */
     authenticateRequest: (request: RequestInformation) => Promise<void>;
+}
+
+
+/**
+ * @interface
+ * An AccessTokenProvider implementation retrieves an access token
+ * to be used by an AuthenticationProvider implementation.
+ * @property {Function} authenticateRequest - The function to authenticate the request.
+ */
+export interface AccessTokenProvider {
+    getAuthorizationToken: (request?: RequestInformation) => Promise<string>;
```

Comment:
Yup. `authorizationToken` and `getAuthorizationToken` should change to accept `url` and `fetchOptions` if they are going through the Graph Core. 

Also for a core only usage, we will have to replace the call to authorization token from the `RequestAdapter` to the `HttpClient` since the core uses the `HTTPClient` and not the `RequestAdapter`. 
Targeting those changes in a separate branch.

Reply:
while I do agree on the replacement of request by url (whatever type it ends up being), I disagree with the fact that fetch options should be passed to that method. Can you elaborate on why you think this would be needed?